### ISSUE TITLE
[core] Resources are getting reported even though they have not changed.

### ISSUE THREAD

**Issue by stephanie-wang** [ray-project/ray#14297]
### What is the problem?

*Ray version and other system information (Python version, TensorFlow version, OS):* v2.0dev

I noticed that resources are getting reported on every heartbeat, even though the raylet is supposed to only report resources if they have changed from the last reported. The culprit seems to be [this line](https://github.com/ray-project/ray/blob/master/src/ray/raylet/node_manager.cc#L460). cc @wumuzi520 

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

Any Ray script.

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by stephanie-wang** [comment]
FYI @wuisawesome @ericl, probably relevant to the planned changes for GCS resource reporting.

**Comment by wuisawesome** [comment]
errr yeah that line is also very race-condition prone. Once GCS pulls from the raylet, we can add a flag to not use lightweight heartbeats for the first request, then get rid of that call.

**Comment by ericl** [comment]
@wuisawesome with long-polling heartbeats from the GCS, we won't need lightweight heartbeats at all right? If nothing changes the polls will simply block until a change occurs.

**Comment by wuisawesome** [comment]
Actually the gcs failure case is a reason why long polling is problematic, but regardless, lightweight heartbeats would still help (for example if only 1 cpu changed, we don't need to re-report all the placement groups on the node)

**Comment by stephanie-wang** [comment]
I don't understand that line at all. Can someone explain why we need it? The comment says it's for GCS failure, but couldn't we just have the GCS ask for new resource reports when it restarts...?

**Comment by wuisawesome** [comment]
> but couldn't we just have the GCS ask for new resource reports when it restarts...?

You need a pull mechanism for that right?

**Comment by stephanie-wang** [comment]
Not necessarily. You can just have the nodes report resources when they detect the GCS failure. If I understand correctly, the problem seems to be that the code is trying to make GCS failures fully transparent, which is not necessary.

**Comment by rkooo567** [comment]
@iycheng Probably related to that the lightweight heartbeat is broken. Are you taking a look at that now?
